interfaces/ldconfig/backend.go

```diff
@@ -59,6 +59,12 @@ func (b *Backend) Setup(appSet *interfaces.SnapAppSet, opts interfaces.Confineme
 			appSet.InstanceName(), err)
 	}
 
+	// Setup ldconfig only of the snap has plugs that require it. For the
+	// moment this is only the system snap.
+	if len(spec.(*Specification).plugs) == 0 {
```

Comment:
I would do this check in the helper, also because it gets the spec casted already